includes/class-signature.php

```diff
@@ -220,15 +105,13 @@ public static function generate_signature( $user_id, $http_method, $url, $date,
 		}
 
 		$signature = null;
-		\openssl_sign( $signed_string, $signature, $key, \OPENSSL_ALGO_SHA256 );
+		\openssl_sign( $signed_string, $signature, $private_key, \OPENSSL_ALGO_SHA256 );
 		$signature = \base64_encode( $signature ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
 
-		$key_id = $user->get_id() . '#main-key';
-
 		if ( ! empty( $digest ) ) {
-			return \sprintf( 'keyId="%s",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="%s"', $key_id, $signature );
+			return \sprintf( 'keyId="%s",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="%s"', $key_id . '#main-key', $signature );
```

Comment:
I think the `#main-key` hash should be part of the key-id

Reply:
Fixed in [487f284](https://github.com/Automattic/wordpress-activitypub/pull/1832/commits/487f28436fb7af0ba31275547dc5074157297c22).